pick seed <= u32::MAX when using metal

rgbkrk opened the pull request:
## Problem

Running the stable diffusion example with Metal results in an error with the seed (possibly) being out of range.

```
$ cargo run --example stable-diffusion --release --features=metal -- --prompt "a cosmonaut on a horse (hd, realistic, high-def)" --sd-version turbo
   Compiling candle-transformers v0.9.1 (/Users/kylekelley/code/src/github.com/huggingface/candle/candle-transformers)
   Compiling candle-examples v0.9.1 (/Users/kylekelley/code/src/github.com/huggingface/candle/candle-examples)
    Finished `release` profile [optimized] target(s) in 13.34s
     Running `target/release/examples/stable-diffusion --prompt 'a cosmonaut on a horse (hd, realistic, high-def)' --sd-version turbo`
Using seed 14094623835642992148
Error: Metal error Metal seed must be less than or equal to u32::MAX

Caused by:
    Metal seed must be less than or equal to u32::MAX
```

The user can manually set `--seed` but we can make this easier.

## Solution

Generate the seed up to `u32::MAX` when metal is enabled.

greenrazer reviewed (changes requested):
Thanks for the contribution!

rgbkrk commented:
Hi @greenrazer! I've addressed your feedback by refactoring the code to use `cfg_if` instead of nested `#[cfg]` attributes. The changes include:

1. Added `cfg-if = "1.0"` to the workspace dependencies in `Cargo.toml`
2. Added `cfg-if` as a dependency in `candle-examples/Cargo.toml`
3. Refactored the seed generation logic in `main.rs` to use the `cfg_if!` macro for better readability

The code now uses `cfg_if!` which is indeed more readable and idiomatic for conditional compilation. Thanks for the suggestion!
